src/mo_types/type.ml

```diff
@@ -67,6 +67,191 @@ and kind =
   | Def of bind list * typ
   | Abs of bind list * typ
 
+(* Efficient comparison *)
+let tag_prim p =
+  match p with
+  | Null -> 0
```

Comment:
Isn't there an OCaml magic builtin to get the raw tag of a sum type?

Reply:
Yeah, but I don't want to use magic.

Reply:
Otherwise this could be interesting
https://www.reddit.com/r/ocaml/comments/mzlsb4/comment/gw1dsqv

Reply:
Yeah, I'm aware there's a ppx for deriving fast comparison functions but I didn't want to depend on that (there's not that much code here). I guess we could replace by that in future, but there's also the possibility that we can do better with hand-rolled code (by choosing the order of comparison for multiple fields, for example).